src/graaflib/algorithm/minimum_spanning_tree.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <graaflib/graph.h>
+#include <graaflib/types.h>
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+namespace graaf::algorithm {
+
+// Helper struct for representing disjoint sets
+struct DisjointSet {
```

Comment:
Could we put this in a namespace `detail`? I think it can also be moved to `minimum_spanning_tree.tpp` to keep the interface in the header file as clean as possible.

Reply:
This I have done